src/components/app/details/triggerView/workflow.service.ts

```diff
@@ -57,6 +58,36 @@ export function processWorkflow(
     let ciPipelineToNodeWithDimension = (ciPipeline: CiPipeline) => ciPipelineToNode(ciPipeline, dimensions)
     const filteredCIPipelines =
         ciResponse?.ciPipelines?.filter((pipeline) => pipeline.active && !pipeline.deleted) ?? []
+    const configuredMaterialList = new Map()
+    for (const ciPipeline of filteredCIPipelines) {
+        configuredMaterialList[ciPipeline.name] = new Set()
+        if (ciPipeline?.ciMaterial?.length > 0) {
+            ciPipeline.ciMaterial.forEach((property, _) =>
+                configuredMaterialList[ciPipeline.name].add(property.gitMaterialId),
+            )
+        } else {
+            ciPipeline.ciMaterial = []
+        }
+    }
+
+    const gitMaterials = ciResponse?.materials ?? []
```

Comment:
Simply & revisit this logic as discussed

Reply:
The simplified logic was not covering on of the corner case and hence breaking, so I put the same logic but extracted it to a function outside to look more streamline